cinder/volume/manager.py

```diff
@@ -680,6 +680,43 @@ def _set_resource_host(self, resource):
             resource.host = volume_utils.append_host(self.host, pool)
             resource.save()
 
+    def _propogate_volume_scheduler_hints(self, context, volume):
+        """Ensure metadata hints are propagated to referenced volumes."""
+        hints = volume_utils.get_scheduler_hints_from_volume(volume)
+        if hints:
+            LOG.debug("Found hints for %(volume)s - %(hints)s",
+                      {'volume': volume.id, 'hints': hints})
+            for hint_key in hints.keys():
```

Comment:
Why do you always use `.keys()` when it's the default behavior of a `dict`? Is there any specific reason to it? It's a little odd for me, so I'd like to understand.

Reply:
probably a remnant of old style